### ISSUE TITLE
Recompile when new file is added

### ISSUE THREAD

**Issue by cveld** [ivogabe/gulp-typescript#236]
The example given in [readme.md](../blob/master/readme.md) does not encompass new files that are added to the project. The watch-task just does not get triggered. 
I found the `gulp-watch` npm package, but I need guidance how I can use it efficiently together with the tsProject structure where all existing files are being cached.

The following fragment is not ideal - it just compiles everything and does not take cached files into account:
```javascript
gulp.task("gulp-watch", function () {
    watch(paths.tscripts.src, batch(function (events, done) {
        events.on('data', console.log);
        gulp.start("typescript", done);
    }));
});
```

**Comment by ivogabe** [comment]
Gulp 4 should be able to detect new files, though that version is not officially released. See #228 for info on how to speed up compilation. In short, type checking requires type information from all files, and thus it's not possible to recompile only one file with type checking. Instead you can give up type checking and use `isolatedModules`.
